### PR TITLE
Added poketch calculator to the gen 4 egg generator context menu

### PR DESCRIPTION
**Pull request by SteveCookTU**

### CODE REVIEW

File: Source/Forms/Gen4/Eggs4.cpp [hunk 1]
```diff
@@ -326,9 +331,32 @@ void Eggs4::seedToTime()
...
+    auto method = static_cast<Method>(ui->comboBoxGeneratorMethod->currentData().toInt());
+    if(method == Method::Gen4Normal)
+    {
+        calcPoketchGenerator->setVisible(true);
+    }
+    else
+    {
+        calcPoketchGenerator->setVisible(false);
+    }
+}
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Eggs4.cpp:362]: You need something similar to this for when the profile changes. I believe poketch is only in DPPt.

File: Source/Forms/Gen4/Eggs4.cpp [hunk 2]
```diff
@@ -326,9 +331,32 @@ void Eggs4::seedToTime()
     time->raise();
 }
 
+void Eggs4::calcPoketch() {
+    QModelIndex index = ui->tableViewGenerator->currentIndex();
+    int advances = generatorModel->data(generatorModel->index(index.row(), 0)).toInt();
+
+    auto *poketch = new Poketch(advances);
+    poketch->show();
+    poketch->raise();
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Eggs4.cpp:340]: Having a dedicated widget for this is probably overkill. This is just a fancy QMessageBox

File: Source/Forms/Gen4/Tools/Poketch.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    connect(ui->pushButtonClose, &QPushButton::clicked, this, &Poketch::on_pushButtonClose_clicked);
+
+    u32 coin = 0;
+    u32 happy = 0;
+    u32 target = 0;
+
+    if(advances < 12)
+    {
+        coin = advances;
+        ui->labelNote->setText("DO NOT SWITCH TO THE HAPPINESS APPLICATION AT ALL");
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Tools/Poketch.cpp:28]: Wrap in a tr() call. Is caps lock necessary?

File: Source/Forms/Gen4/Tools/Poketch.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    } else
+    {
+        target = advances - 12;
+        happy = target / 12;
+        coin = target % 12;
+
+        if(happy == 0)
+        {
+            ui->labelNote->setText("SWITCH TO THE HAPPINESS APPLICATION ONCE BUT DO NOT CLICK");
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Tools/Poketch.cpp:37]: Wrap in a tr() call. Is caps lock necessary?

**Review by Admiral-Fish — CHANGES_REQUESTED** [review]

File: Source/Forms/Gen4/Tools/Poketch.cpp (hunk 3 above)
Comment (SteveCookTU) [reply Source/Forms/Gen4/Tools/Poketch.cpp:28]: Caps lock just cause it was in Reporter lol fixed though

File: Source/Forms/Gen4/Eggs4.cpp (hunk 2 above)
Comment (SteveCookTU) [reply Source/Forms/Gen4/Eggs4.cpp:340]: Did this for design purposes, easier to layout with the multiple outputs and the extra note depending on the advances. Easily up for debate to switch but want to discuss before switching it over

File: Source/Forms/Gen4/Eggs4.cpp (hunk 1 above)
Comment (Admiral-Fish) [reply Source/Forms/Gen4/Eggs4.cpp:372]: This also needs to be profile aware. Only update if profile is DPPt.

File: Source/Forms/Gen4/Eggs4.cpp [hunk 5]
```diff
@@ -326,9 +341,32 @@ void Eggs4::seedToTime()
...
+
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Eggs4.cpp:351]: nit: whitespace

File: Source/Forms/Gen4/Tools/Poketch.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+Poketch::Poketch(u32 advances, QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::Poketch)
+{
+    ui->setupUi(this);
+    setAttribute(Qt::WA_QuitOnClose, false);
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    setupModels(advances);
+
```
Comment (Admiral-Fish) [inline Source/Forms/Gen4/Tools/Poketch.cpp:14]: nit: whitespace

File: Source/Forms/Gen4/Eggs4.cpp (hunk 2 above)
Comment (Admiral-Fish) [reply Source/Forms/Gen4/Eggs4.cpp:350]: Whichever you think is best. It just seemed like a fancy QMessageBox to me

**Review by Admiral-Fish — CHANGES_REQUESTED** [review]
Just noticed there are some style issues. I have a .clang-format file, please run it against your changes.

File: Source/Forms/Gen4/Eggs4.cpp (hunk 1 above)
Comment (SteveCookTU) [reply Source/Forms/Gen4/Eggs4.cpp:372]: This was added in the second to last commit

**Pull request merged**